packages/blade/src/utils/hasSameObjectStructure/__tests__/hasSameObjectStructure.test.ts

```diff
@@ -0,0 +1,49 @@
+import { hasSameObjectStructure } from '../hasSameObjectStructure';
+
+describe('hasSameObjectStructure', () => {
+  it('should return true for objects with the same structure', () => {
+    const obj1 = {
+      a: {
+        b: {
+          c: 1,
+        },
+        d: 'hello',
```

Comment:
in extra overrides we need subset of the actual structure right? basically we need whatever the structure is present that shall be the same but not exact same structure right? because extraOverrides is always subset of the theme structure and not same

Reply:
Overriding with brand colors will keep the structure same but will change the values at leaf nodes, which will mean the new theme is now not equal to payment or banking theme. So when you try to override a branded theme with extra overrides, overrideTheme will stop you from doing so as per current logic